### PR TITLE
Notifcation branch

### PR DESCRIPTION
**Pull request by DenisGauthier**
I now have the proper method set up - Not a class whoops in my terminology. This allows you to call notifications with either close to due or past due and task name. The first instance of the call will create a pop-up asking for permission, If granted the notification setting is turned on automatically (No more going into app info and settings, woot!) and should only ask the once.

### CODE REVIEW

**Comment by DenisGauthier** [comment]
Also as previously noticed by Justin in my last merge - I am working towards using automatic alert notifications, currently, though I need to be more up-to-date with the main to make sure the method would work.  However, I do plan to make it automatic this weekend.

File: app/src/main/java/com/example/todolist/MainActivity.java [hunk 1]
```diff
@@ -67,14 +104,28 @@ public class MainActivity extends AppCompatActivity implements ToDoClickListener
...
+        //Creates the notification channel that can be toggled on in the app info settings - Default is off.
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
+        {
+            NotificationChannel channel = new NotificationChannel("NotifyLate", "Late Task Notification", NotificationManager.IMPORTANCE_HIGH);
+
+            NotificationManager manager = getSystemService(NotificationManager.class);
+            manager.createNotificationChannel(channel);
+        }
+
+        recyclerView = findViewById(R.id.recyclerView);
```
Comment (justino599) [inline app/src/main/java/com/example/todolist/MainActivity.java:124]: This is an error. No such variable "recyclerView". There is also no component "R.id.recyclerView". Seems that just removing it fixes it and causes no problems

**Review by justino599 — CHANGES_REQUESTED** [review]
Looking good! I still think we should be adding a way for the user to create and schedule notifications though. Can add another menu item to the power menu that opens with the three dots. The place to do that is in onEditClick(). Maybe just open up a new activity where they can either turn on or off notifications for the given task and maybe set how long before the due date they want to receive a notification. If you want some help figuring out how to use the power menu or creating a new activity, just shoot me a message and we can hop on a call to figure it out.

File: app/src/main/java/com/example/todolist/MainActivity.java (hunk 1 above)
Comment (DenisGauthier) [reply app/src/main/java/com/example/todolist/MainActivity.java]: kk removed that line, I wasnt getting that error on myside. let me know if that corrected it for you!

**Review by justino599 — APPROVED** [review]

**Pull request merged**